File: src/main/java/xyz/lncvrt/galaxyboxpvp/PlaceholderAPIExpansion.java
package xyz.lncvrt.galaxyboxpvp;

import com.booksaw.betterTeams.Team;
import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.ChatColor;
import org.bukkit.Statistic;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class PlaceholderAPIExpansion extends PlaceholderExpansion {
    @Override
    public @NotNull String getIdentifier() {
        return "galaxyboxpvp";
    }

    @Override
    public @NotNull String getAuthor() {
        return "Lncvrt";
    }

    @Override
    public @NotNull String getVersion() {
        return "1.0.0";
    }

    @Override
    public boolean persist() {
        return true;
    }

    @Override
    public boolean canRegister() {
        return true;
    }

    @Override
    public @Nullable String onPlaceholderRequest(Player player, @NotNull String identifier) {
        switch (identifier) {
            case "hearts":
                return String.format("%.1f", player.getHealth() / 2);
            case "kills":
                return String.valueOf(player.getStatistic(Statistic.PLAYER_KILLS));
            case "deaths":
                return String.valueOf(player.getStatistic(Statistic.DEATHS));
            case "kdr":
                int kills = player.getStatistic(Statistic.PLAYER_KILLS);
                int deaths = player.getStatistic(Statistic.DEATHS);
                return deaths == 0 ? String.valueOf(kills) : String.format("%.1f", (double) kills / deaths);
            case "team":
                Team team1 = Team.getTeam(player);
                if (team1 == null) return "";
                return ChatColor.translateAlternateColorCodes('&', "%s[%s%s%s%s]%s ".formatted(ChatColor.GRAY, ChatColor.RESET, team1.getDisplayName(), ChatColor.RESET, ChatColor.GRAY, ChatColor.RESET));
            case "teamname":
                Team team2 = Team.getTeam(player);
                if (team2 == null) return "None";
                return team2.getName();
            default:
                return null;
        }
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/PlaceholderAPIExpansion.kt
@file:Suppress("DEPRECATION")

package xyz.lncvrt.galaxyboxpvp

import com.booksaw.betterTeams.Team
import me.clip.placeholderapi.expansion.PlaceholderExpansion
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import org.bukkit.ChatColor
import org.bukkit.Statistic
import org.bukkit.entity.Player
import org.sayandev.stickynote.bukkit.utils.AdventureUtils.legacyColored
import org.sayandev.stickynote.bukkit.utils.AdventureUtils.legacyString
import javax.annotation.Nullable

class PlaceholderAPIExpansion : PlaceholderExpansion() {
    override fun getIdentifier(): String {
        return "galaxyboxpvp"
    }

    override fun getAuthor(): String {
        return "Lncvrt"
    }

    override fun getVersion(): String {
        return "1.0.0"
    }

    override fun persist(): Boolean {
        return true
    }

    override fun canRegister(): Boolean {
        return true
    }

    @Nullable
    override fun onPlaceholderRequest(player: Player, identifier: String): String? {
        when (identifier) {
            "hearts" -> return String.format("%.1f", player.health / 2)
            "kills" -> return player.getStatistic(Statistic.PLAYER_KILLS).toString()
            "deaths" -> return player.getStatistic(Statistic.DEATHS).toString()
            "kdr" -> {
                val kills = player.getStatistic(Statistic.PLAYER_KILLS)
                val deaths = player.getStatistic(Statistic.DEATHS)
                return if (deaths == 0) kills.toString() else String.format("%.1f", kills.toDouble() / deaths)
            }
            "team" -> {
                val team1: Team? = Team.getTeam(player)
                if (team1 == null) return ""
                return LegacyComponentSerializer.legacyAmpersand().serialize(MiniMessage.miniMessage().deserialize("<gray>[</gray>%s<gray>]</gray>".format(team1.displayName)))
            }
            "teamname" -> {
                val team2: Team? = Team.getTeam(player)
                if (team2 == null) return "None"
                return team2.name
            }
            else -> return null
        }
    }
}